### PR TITLE
Improve docs for embed-bitcode and linker-plugin-lto

### PR DESCRIPTION
**Pull request by alexcrichton**
Follow-up from #71716 I wasn't able to add in time.

### CODE REVIEW

File: src/doc/rustc/src/codegen-options/index.md [hunk 1]
```diff
@@ -62,24 +62,42 @@ the linker.
...
-rlibs. It takes one of the following values:
+This flag controls whether or not the compiler embeds LLVM bitcode into object
+files. It takes one of the following values:
 
 * `y`, `yes`, `on`, or no value: put bitcode in rlibs (the default).
 * `n`, `no`, or `off`: omit bitcode from rlibs.
 
-LLVM bitcode is only needed when link-time optimization (LTO) is being
-performed, but it is enabled by default for backwards compatibility reasons.
+The default value of this option is "yes".
```
Comment (nnethercote) [inline src/doc/rustc/src/codegen-options/index.md:71]: Remove this sentence, it's covered by "(the default)" above, and other boolean options with static defaults don't have a sentence like this.

File: src/doc/rustc/src/codegen-options/index.md [hunk 2]
```diff
@@ -62,24 +62,42 @@ the linker.
...
+(e.g. if you're not compiling for iOS or you're not performing LTO). For these
+reasons, Cargo uses `-C embed-bitcode=no` whenever possible. Likewise, if you
+are building directly with `rustc` we recommend using `-C embed-bitcode=no`
+whenever you are not using LTO.
 
 If combined with `-C lto`, `-C embed-bitcode=no` will cause `rustc` to abort
 at start-up, because the combination is invalid.
 
+> **Note**: if you're building Rust code with LTO then you probably don't even
+> need the `embed-bitcode` option turned on. You'll likly want to use
```
Comment (nnethercote) [inline src/doc/rustc/src/codegen-options/index.md:90]: s/likly/likely

File: src/doc/rustc/src/codegen-options/index.md [hunk 3]
```diff
@@ -62,24 +62,42 @@ the linker.
...
+> `-Clinker-plugin-lto` instead which skips generating object files entirely and
+> simply replaces object files with LLVM bitcode. The only purpose for
+> `-Cembed-bitcode` is when you're generating an rlib that is both being used
+> with and without LTO. For example Rust's standard library ships with embedded
+> bitcode since users link to it both with and without LTO.
+>
+> This also may make you wonder why the default is "yes" for this option. The
```
Comment (nnethercote) [inline src/doc/rustc/src/codegen-options/index.md:97]: s/"yes"/`yes`/

**Review by nnethercote — CHANGES_REQUESTED** [review]
r=me once the nits are addressed.

File: src/doc/rustc/src/codegen-options/index.md (hunk 1 above)
Comment (alexcrichton) [reply src/doc/rustc/src/codegen-options/index.md:71]: Whoops, missed that!

**Pull request merged**